interp/constrintern.ml

```diff
@@ -1260,14 +1260,26 @@ let glob_sort_of_level (level: glob_level) : glob_sort =
   | UAnonymous {rigid} -> UAnonymous {rigid}
   | UNamed id -> UNamed [id,0]
 
+let pr_constant c = pr_qualid (Nametab.shortest_qualid_of_global Id.Set.empty c)
+
+let warn_deprecated_constant =
+  Deprecation.create_warning ~object_name:"Constant" ~warning_name:"deprecated-constant"
+    pr_constant
+
 (* Is it a global reference or a syntactic definition? *)
 let intern_qualid ?(no_secvar=false) qid intern env ntnvars us args =
   let loc = qid.loc in
   match intern_extended_global_of_qualid qid with
   | TrueGlobal (GlobRef.VarRef _) when no_secvar ->
       (* Rule out section vars since these should have been found by intern_var *)
       raise Not_found
-  | TrueGlobal ref -> (DAst.make ?loc @@ GRef (ref, us)), Some ref, args
+  | TrueGlobal ref ->
+    begin match ref with
+    | GlobRef.ConstRef c ->
+      Option.iter (fun d -> warn_deprecated_constant (ref,d)) (Deprecation.lookup_constant_deprecation c);
```

Comment:
I wonder if this is the right place to warn
ie if I do `Existing Instance some_deprecated_constant` it won't go through this code.

Reply:
Right, there's this and a few other ways we can depend on a deprecated constant and escape the warnings (if it has been inserted by unification, for example).

So where would you put the warning emitter?

Reply:
Probably Nametab.locate_extended
probably merge the syndef warning with it